kubeflow/pull-upstream.sh

```diff
@@ -1,12 +1,26 @@
 #!/usr/bin/env bash
+#
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
 set -ex
 
 export KUBEFLOW_MANIFESTS_VERSION=v1.3.0
 export KUBEFLOW_MANIFESTS_REPO=https://github.com/kubeflow/manifests.git
 
-export KUBEFLOW_PIPELINES_VERSION=1.5.0
-export KUBEFLOW_PIPELINES_REPO=https://github.com/kubeflow/pipelines.git
+# Pull Kubeflow Pipelines upstream manifests.
+./apps/pipelines/pull-upstream.sh
```

Comment:
Should we use `make -C apps/pipelines pull-upstream` instead?

Reply:
This is a good point. I am not sure. If we add it to makefile the model is more consistent with other commands.

If use separate pull-upstream.sh, packages without makefile can easily add them too.

Reply:
The reason I ask is that you have already created a pull-upstream make target in https://github.com/kubeflow/gcp-blueprints/blob/2b8e52dd10796d78e40f40d6d9617b9e64b773b3/kubeflow/apps/pipelines/Makefile#L73, so I guess the goal is to use it. 

As mentioned in the other comment, if the eventual state is to separate pull-upstream commands to each sub-folder, or use kpt subpackage when it becomes available, I prefer to create a template script for pull-upstream, so each sub-folder can call this template from Makefile, instead of having pull-upstream.sh in each sub-folder. Because it will increase amount of files in the whole repo, and it is hard to keep them consistent. If there is no Makefile, then we will use the default template to pull upstream.

For this PR, feel free to keep the current usage. And we can update them at one go when we break down the kubeflow/pull-upstream.sh.

Reply:
Yes, I added the makefile rule for consistency.

> If there is no Makefile, then we will use the default template to pull upstream.

I think we need to know where upstream is and which version, these info need to be somewhere in the folder.

Reply:
Logged in https://github.com/kubeflow/gcp-blueprints/issues/287